nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,631 @@
+            if is_classification:
+                best_y['inverse'] = inverse_prob
+
+    if select_features is not None:
+        best_coef = select_features.inverse_transform(best_coef)
+    # if sklearn.__version__ > 1.7
+    if isinstance(estimator, SVR):
+        best_coef = -best_coef
+
+    return pos_label, best_coef, best_intercept, best_y, best_params
+
+
+def _check_masking(mask, smoothing_fwhm, target_affine, target_shape,
+                   standardize, mask_strategy, memory, memory_level):
+    """Setup a nifti masker."""
+    # mask is an image, not a masker
+    if not isinstance(mask, (NiftiMasker, MultiNiftiMasker)):
+        masker = NiftiMasker(mask_img=mask,
+                             smoothing_fwhm=smoothing_fwhm,
+                             target_affine=target_affine,
+                             target_shape=target_shape,
+                             standardize=standardize,
+                             mask_strategy=mask_strategy,
+                             memory=memory,
+                             memory_level=memory_level, )
+    # mask is a masker object
+    else:
+        try:
+            if hasattr(mask, 'mask_img_'):
+                mask_img = mask.mask_img_
+            masker = clone(mask)
+            masker.mask_img_ = mask_img
+        except TypeError as e:
+            # Workaround for a joblib bug: in joblib 0.6, a Memory object
+            # with cachedir = None cannot be cloned.
+            masker_memory = mask.memory
+            if masker_memory.cachedir is None:
+                mask.memory = None
+                masker = clone(mask)
+                mask.memory = masker_memory
+                masker.memory = Memory(cachedir=None)
+            else:
+                # The error was raised for another reason
+                raise e
+
+        for param_name in ['target_affine', 'target_shape',
+                           'smoothing_fwhm', 'mask_strategy',
+                           'memory', 'memory_level']:
+            if getattr(masker, param_name) is not None:
+                warnings.warn('Parameter %s of the masker overriden'
+                              % param_name)
+                setattr(masker, param_name, locals()[param_name])
+        if getattr(masker, 'mask_img_') is not None:
+            warnings.warn('The mask_img_ of the masker will be copied')
+    return masker
+
+
+def _check_param_grid(estimator, X, y, param_grid):
+    """Check param_grid and return sensible default if none is given.
+    """
+    if param_grid is None:
+        param_grid = {}
+        # define loss function
+        if isinstance(estimator, LogisticRegression):
+            loss = 'log'
+        elif isinstance(estimator, (LinearSVC, _BaseRidge, SVR)):
+            loss = 'l2'
+        min_c = l1_min_c(X, y, loss=loss) \
+            if hasattr(estimator, 'penalty') \
+            and estimator.penalty == 'l1' else .5
+        param_grid['C'] = np.array([2, 20, 200]) * min_c
```

Comment:
Where does this heuristic for the parameter grid come from? There should be an informatic comment on that.